components/log_wrappers/src/proto.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
```

Comment:
How about moving this to protobuf-build crate instead? So developer doesn't have to remember using `ProtobufValue` instead when logging values. And the implement can be reused by other projects like tikv client.

Reply:
I think there's some pitfalls on moving this part to `protobuf-build`

* `REDACT_INFO_LOG` atomic bool will become part of the generated protobuf source code
* After `proto.rs` is moved to `protobuf-build`
    - Developers will have to specify what field to redact in `protobuf-build` instead of in TiKV
    - `protobuf-build` will contain structures (e.g. `Region`, `xxxCmd`) inside protobuf spec, and `protobuf-build` can only be used with TiKV protobuf specs.

This seems very unsound...

Reply:
> `REDACT_INFO_LOG` atomic bool will become part of the generated protobuf source code

Indeed, and in such way can other project reuse the implements.

> REDACT_INFO_LOG atomic bool will become part of the generated protobuf source code

In fact, what I suggested is adapting PbPrint to make it adjustable at runtime instead of moving the file to protobuf-build.

The most general way is protocol authors define what fields need to be redacted. However, to make it simple, just redacting all bytes/strings should be enough.

Reply:
Alright. Then I suggest the following migration strategy:

* In `protobuf-build`, add `REDACT_BYTES` and modify `PbPrint`, so that all data printed with `PbPrint` will redact bytes and strings fields.
* close (or remove most of the contents) in this PR, and let TiKV to set `REDACT_BYTES` in `protobuf-build` if TiKV config `REDACT_INFO_LOG` is set.

If OK, I'll work on the above migration this week.

Reply:
Agree this should be move to protobuf side, since we plan to update `PbPrint` anyway. I think we can have another `REDACT_INFO_LOG` atomic bool on protobuf side, which control redaction of protobuf structs only.

For protobuf, is it possible to implement annotation in protobuf for redaction? like
```
message Region {
    uint64 id = 1;
    bytes start_key = 2 [need_redact = true];
    bytes end_key = 3 [need_redact = true];
}
```

Reply:
Yes, it's possible. To make things simple, I suggest to just hide all bytes at first and leave it for future plan (perhaps a task for community).

Reply:
ok